react_juce/core/TextShadowView_Yoga.cpp

```diff
@@ -0,0 +1,49 @@
+#include "TextShadowView.h"
+#include "YogaImplInclude.cpp"
+
+namespace reactjuce
+{
+    //==============================================================================
+    /** We use this method to measure the size of a given string so that the
+     *  text container knows what size to take.
+     */
+    YGSize measureTextNode(YGNodeRef node, float width, YGMeasureMode widthMode, float height, YGMeasureMode heightMode)
+    {
+        juce::ignoreUnused(widthMode);
+        juce::ignoreUnused(height);
+        juce::ignoreUnused(heightMode);
+
+        auto context = reinterpret_cast<TextShadowView*>(YGNodeGetContext(node));
+        auto view = dynamic_cast<TextView*>(context->getAssociatedView());
+
+        jassert (view != nullptr);
+
+        // TODO: This is a bit of an oversimplification. We have a YGMeasureMode which
+        // is one of three things, "undefined", "exact", or "at-most." Here we're kind of
+        // just ignoring that, and in cases like `white-space: nowrap;` we want to ignore it,
+        // but it would probably be good to get specific for each case.
+        // See https://github.com/facebook/yoga/pull/576/files
+        auto tl = view->getTextLayout(width);
+
+        return { tl.getWidth(), tl.getHeight() };
+    }
+
+    //==============================================================================
+    class TextShadowView::Pimpl
+    {
+    public:
+        Pimpl()
+        {
+            YGNodeSetContext(yogaNode, this);
```

Comment:
@nick-thompson , here is a thing to be decided. How to get access to the relevant `yogaNode` instance inside the pimpl. We can pass the pimpl's owning `TextShadowView` instance through to the `Pimpl` constructor and then have this `Pimpl` again as a friend to `ShadowView/TextShadowView`. Perhaps there is a better design to use here though? 

I'm actually not totally against `friend` in these particular cases even though I think it should almost always be considered evil. What do you think?